CondTools/CTPPS/plugins/CTPPSPixGainCalibsESAnalyzer.cc

```diff
@@ -0,0 +1,151 @@
+    detId_readable[2023227392]="Arm_0_Sec_45_St_2_Pot_3_Plane_0";
+    detId_readable[2023292928]="Arm_0_Sec_45_St_2_Pot_3_Plane_1";
+    detId_readable[2023358464]="Arm_0_Sec_45_St_2_Pot_3_Plane_2";
+    detId_readable[2023424000]="Arm_0_Sec_45_St_2_Pot_3_Plane_3";
+    detId_readable[2023489536]="Arm_0_Sec_45_St_2_Pot_3_Plane_4";
+    detId_readable[2023555072]="Arm_0_Sec_45_St_2_Pot_3_Plane_5";
+    detId_readable[2031616000]="Arm_1_Sec_56_St_0_Pot_3_Plane_0";
+    detId_readable[2031681536]="Arm_1_Sec_56_St_0_Pot_3_Plane_1";
+    detId_readable[2031747072]="Arm_1_Sec_56_St_0_Pot_3_Plane_2";
+    detId_readable[2031812608]="Arm_1_Sec_56_St_0_Pot_3_Plane_3";
+    detId_readable[2031878144]="Arm_1_Sec_56_St_0_Pot_3_Plane_4";
+    detId_readable[2031943680]="Arm_1_Sec_56_St_0_Pot_3_Plane_5";
+    detId_readable[2040004608]="Arm_1_Sec_56_St_2_Pot_3_Plane_0";
+    detId_readable[2040070144]="Arm_1_Sec_56_St_2_Pot_3_Plane_1";
+    detId_readable[2040135680]="Arm_1_Sec_56_St_2_Pot_3_Plane_2";
+    detId_readable[2040201216]="Arm_1_Sec_56_St_2_Pot_3_Plane_3";
+    detId_readable[2040266752]="Arm_1_Sec_56_St_2_Pot_3_Plane_4";
+    detId_readable[2040332288]="Arm_1_Sec_56_St_2_Pot_3_Plane_5";
+
+  }
+  explicit  CTPPSPixGainCalibsESAnalyzer(int i) {
+    std::cout<<"CTPPSPixGainCalibsESAnalyzer "<<i<<std::endl; 
+    detId_readable[2014838784]="Arm_0_Sec_45_St_0_Pot_3_Plane_0";
+    detId_readable[2014904320]="Arm_0_Sec_45_St_0_Pot_3_Plane_1";
+    detId_readable[2014969856]="Arm_0_Sec_45_St_0_Pot_3_Plane_2";
+    detId_readable[2015035392]="Arm_0_Sec_45_St_0_Pot_3_Plane_3";
+    detId_readable[2015100928]="Arm_0_Sec_45_St_0_Pot_3_Plane_4";
+    detId_readable[2015166464]="Arm_0_Sec_45_St_0_Pot_3_Plane_5";
+    detId_readable[2023227392]="Arm_0_Sec_45_St_2_Pot_3_Plane_0";
+    detId_readable[2023292928]="Arm_0_Sec_45_St_2_Pot_3_Plane_1";
+    detId_readable[2023358464]="Arm_0_Sec_45_St_2_Pot_3_Plane_2";
+    detId_readable[2023424000]="Arm_0_Sec_45_St_2_Pot_3_Plane_3";
+    detId_readable[2023489536]="Arm_0_Sec_45_St_2_Pot_3_Plane_4";
+    detId_readable[2023555072]="Arm_0_Sec_45_St_2_Pot_3_Plane_5";
+    detId_readable[2031616000]="Arm_1_Sec_56_St_0_Pot_3_Plane_0";
+    detId_readable[2031681536]="Arm_1_Sec_56_St_0_Pot_3_Plane_1";
+    detId_readable[2031747072]="Arm_1_Sec_56_St_0_Pot_3_Plane_2";
+    detId_readable[2031812608]="Arm_1_Sec_56_St_0_Pot_3_Plane_3";
+    detId_readable[2031878144]="Arm_1_Sec_56_St_0_Pot_3_Plane_4";
+    detId_readable[2031943680]="Arm_1_Sec_56_St_0_Pot_3_Plane_5";
+    detId_readable[2040004608]="Arm_1_Sec_56_St_2_Pot_3_Plane_0";
+    detId_readable[2040070144]="Arm_1_Sec_56_St_2_Pot_3_Plane_1";
+    detId_readable[2040135680]="Arm_1_Sec_56_St_2_Pot_3_Plane_2";
+    detId_readable[2040201216]="Arm_1_Sec_56_St_2_Pot_3_Plane_3";
+    detId_readable[2040266752]="Arm_1_Sec_56_St_2_Pot_3_Plane_4";
+    detId_readable[2040332288]="Arm_1_Sec_56_St_2_Pot_3_Plane_5";
+  }
```

Comment:
It looks the two constructors share the same hard-coded data. I'd recommend to implement one based on the other one...